mbed-client-mbedtls/m2mconnectionsecuritypimpl.h

```diff
@@ -24,7 +24,10 @@
 extern "C"{
 #include "pal_TLS.h"
 }
-
+#ifdef USE_CUSTOM_MBEDTLS_ENTROPY
```

Comment:
The IFDEF should be renamed so that it's matching the rest of the MBED_CLOUD_CLIENT configs AND also the config should be added to the mbed-cloud-client-example config.